tests/integration_test_net.go

```diff
@@ -289,6 +357,11 @@ func startIntegrationTestNet(
 	// the network's session needs to know about the network itself
 	net.net = net
 
+	if testProfileEnabled != nil && *testProfileEnabled {
+		net.profiler = netProfiler{tb: t}
+		go profileRoutine(t.Context(), &net.profiler)
```

Comment:
The member field is pointless. You could just pass the `t` reference to the `profileRoutine` function and track the `highSeen` as a local variable inside the `profileRoutine`.

Reply:
struct removed, only `t` pass and `highSeen` is now just a local variable